edocumentation-transfer-obj/src/main/java/org/dcsa/edocumentation/transferobjects/validation/UtilizedTransportEquipmentValidator.java

```diff
@@ -0,0 +1,34 @@
+package org.dcsa.edocumentation.transferobjects.validation;
+
+import org.dcsa.edocumentation.transferobjects.EquipmentTO;
+import org.dcsa.edocumentation.transferobjects.UtilizedTransportEquipmentTO;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Objects;
+
+public class UtilizedTransportEquipmentValidator
+    implements ConstraintValidator<ValidUtilizedTransportEquipment, UtilizedTransportEquipmentTO> {
+
+  @Override
+  public void initialize(ValidUtilizedTransportEquipment constraintAnnotation) {
+    ConstraintValidator.super.initialize(constraintAnnotation);
+  }
+
+  @Override
+  public boolean isValid(
+      UtilizedTransportEquipmentTO utilizedTransportEquipmentTO,
+      ConstraintValidatorContext constraintValidatorContext) {
+    if (utilizedTransportEquipmentTO != null) {
+      if (utilizedTransportEquipmentTO.isShipperOwned()) {
+        return containsWeightAndUnit(utilizedTransportEquipmentTO.equipment());
+      }
+      return true;
+    }
+    return false;
+  }
+
+  private Boolean containsWeightAndUnit(EquipmentTO equipment) {
```

Comment:
```suggestion
  private boolean containsWeightAndUnit(EquipmentTO equipment) {
```

It cannot return `null`.

Reply:
fixed